### ISSUE TITLE
In image-based filtering, only imagenet L14 is supported, making extracting B32 embeddings useless

### ISSUE THREAD

**Issue by zwsjink** [mlfoundations/datacomp#50]
![image](https://github.com/mlfoundations/datacomp/assets/8237953/1849acfa-9b00-4a07-aeda-a29386191435)
Well, when I go through the apply_filter.py source code, I spot that in image-based filtering step, only in1k_clip_vit_l14_{i} is provided. I suppose the dimension of the datacomp embeddings/centroids should align with imagenet ? If so, extracting B-32 (which is in 512 ) seems to be useless?

**Comment by sagadre** [comment]
Hi @zwsjink! For the main image-based filtering baseline in the paper, we used L/14 features, so to replicate this baseline, only the L/14 features are needed. B/32 features are used for other baselines (e.g., [clip score filtering baselines](https://github.com/mlfoundations/datacomp/blob/dca0fb8ae6d4efeb4b32d52c89cd351cdcfe418c/README.md#clip-score-filtering)). Hope that answers the question, but feel free to let me know if not!

**Comment by zwsjink** [comment]
> Hi @zwsjink! For the main image-based filtering baseline in the paper, we used L/14 features, so to replicate this baseline, only the L/14 features are needed. B/32 features are used for other baselines (e.g., [clip score filtering baselines](https://github.com/mlfoundations/datacomp/blob/dca0fb8ae6d4efeb4b32d52c89cd351cdcfe418c/README.md#clip-score-filtering)). Hope that answers the question, but feel free to let me know if not!

Thanks for the clarification. Yeah, I do see in clip-score based filter step, both l14 and b32 are supported. I suppose you guys have tried image-based filtering on both L14 and B32  embeddings and find that L14 outperform B32?

**Comment by zwsjink** [comment]
> > Hi @zwsjink! For the main image-based filtering baseline in the paper, we used L/14 features, so to replicate this baseline, only the L/14 features are needed. B/32 features are used for other baselines (e.g., [clip score filtering baselines](https://github.com/mlfoundations/datacomp/blob/dca0fb8ae6d4efeb4b32d52c89cd351cdcfe418c/README.md#clip-score-filtering)). Hope that answers the question, but feel free to let me know if not!
> 
> Thanks for the clarification. Yeah, I do see in clip-score based filter step, both l14 and b32 are supported. I suppose you guys have tried image-based filtering on both L14 and B32 embeddings and find that L14 outperform B32?

Just want to add my observation here,  by going through the data provided in Table21-24 in the paper,  there is not too much benefit when moving from B32 to L14 in clip score thresholding, at least in small/medium/large . So that's why I'm wondering if it worth using L14  for  both clip score thresholding and image-based filtering.  Probably, B32 is enough for me to get an acceptable accuracy  and more compute/storage resource friendly

**Comment by sagadre** [comment]
Hi @zwsjink I agree with your conclusion! And great point about the small differences between B/32 and L/14 features for filtering and clustering. We were indeed surprised that using a stronger CLIP backbone for clustering/filtering did not lead to large gains in downstream performance. 

This brings up interesting questions related to what makes a good dataset filtering model, which at least from this comparison, seems slightly different than what makes a good zero-shot model.

**Comment by zwsjink** [comment]
@sagadre  I also spot that compared with  clip-score thresholding,  image-based filter does not bring too much benefit.
![image](https://github.com/mlfoundations/datacomp/assets/8237953/648106df-fbfa-4dce-9dce-475934445eda)
is there a reason for you guys  using Imagenet to do a relevance filter here?

**Comment by sagadre** [comment]
While the difference between clip score filtering and image-based filtering is within a percentage point (pp), we found it interesting that “stacking” these filtering methods produce a substantial gain (approx. 3pp). There is definitely more here to understand (i.e., when one should stack filtering methods). 

We experimented with IN1k and IN21k for image-based filtering just because these are common datasets and seemed like reasonable baselines. Investing additional datasets for filtering is also an interesting direction!
